### DISCUSSION TITLE
Workflow is suddenly triggering an extra job

### DISCUSSION THREAD

**Discussion by octaviocorona** [community/community#25135] [Actions]
<pre><code>name: Build
on: [push,pull_request]

jobs:
  job_1:
    if: github.repository == 'repo/A'
    runs-on: self-hosted

job_2:
if: github.repository == 'repo/A'
runs-on: self-hosted
job_3:
if: github.repository == 'repo/B'
runs-on: self-hosted
</code></pre>
<p>Hello,</p>
<p>I’ve implemented a workflow in our repository that would run based on what repository we were pushing into. For example, if we were pushing from ‘repo/A’ then job_1 and job_2 would execute. As of 4/3/2020, I started to see that job_3 would attempt to execute after jobs 1 and 2, but fails immediately.</p>
<p>From the perspective of a PR I’ve noticed that the <em>checks</em> would fail 4/6:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/c352f723e8d92cb22b65151f68a3754da0cdab4a.png" data-download-href="/uploads/short-url/rRURh1YmRGSEpxZDTVtEsitgdt0.png?dl=1" title="Screen Shot 2020-04-06 at 9.32.09 AM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24599782/181089091-229683c2-a7d8-4c1c-8d50-1741b9ee891c.png" alt="Screen Shot 2020-04-06 at 9.32.09 AM.png" data-base62-sha1="rRURh1YmRGSEpxZDTVtEsitgdt0" width="690" height="257" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/c352f723e8d92cb22b65151f68a3754da0cdab4a_2_690x257.png, https://aws1.discourse-cdn.com/github/optimized/2X/c/c352f723e8d92cb22b65151f68a3754da0cdab4a_2_1035x385.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/c/c352f723e8d92cb22b65151f68a3754da0cdab4a.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/c352f723e8d92cb22b65151f68a3754da0cdab4a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-04-06 at 9.32.09 AM.png</span><span class="informations">1260×470 21.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>but from the Actions tab perspective we see that the GitHub Actions is successful (Note: job_4 is job_3):</p>
<p><img src="https://user-images.githubusercontent.com/24599782/181089092-022b216f-e92e-4538-a38a-54e8835073ff.png" alt="Screen Shot 2020-04-06 at 9.33.51 AM.png" data-base62-sha1="q90wM7An5sLv9ZpuisCcArgleqb" width="318" height="128"> <img src="https://user-images.githubusercontent.com/24599782/181089093-bfb6cfaa-fcce-423a-b059-d69b822cee54.png" alt="Screen Shot 2020-04-06 at 9.34.40 AM.png" data-base62-sha1="szO1JSzGDZ3xOlTQcdIFUPnNpz1" width="432" height="500"></p>
<p>Any information would be much appreciated!</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/octaviocorona">@octaviocorona</a> ,</p>
<p>Do you mean according to the if conditional, the <strong>job_3</strong> should be skipped to run and not listed in the check list of the PR, but now it is listed on the PR and shows as failed to run? Any misunderstanding, feel free to tell me.</p>
<p>Yes, the skipped jobs generally will not be listed in the workflow run.</p>
<blockquote>
<p>From the perspective of a PR I’ve noticed that the <em>checks</em> would fail 4/6:</p>
</blockquote>
<blockquote>
<p>but from the Actions tab perspective we see that the GitHub Actions is successful (Note: job_4 is job_3):</p>
</blockquote>
<p>But I see from the screenshots you provided, the <strong>job_3</strong> is showing as failed both in the check list and in the workflow run on the <strong>Actions</strong> tab. The workflow run is not successful.</p>
<p>If possible, please share your repository with us, so that we can check more detailed configurations in your workflow YAML file and more detailed info about the workflow run for further investigation and evaluation.</p>

**Comment by octaviocorona** [comment]
<p>Yes you are correct in regards to <strong>job_3</strong> should be skipped to run and not listed in the check list, but it’s now listed. The repository that contains the <strong>Actions</strong> error is on a private repository. Here’s the public listing of the repository that utilizes the same <strong>main.yml</strong> : <a href="https://github.com/BugSwarm/bugswarm" rel="nofollow noopener">https://github.com/BugSwarm/bugswarm</a> .</p>
<p>Thanks!</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/octaviocorona">@octaviocorona</a> ,</p>
<p>I noticed a recent reported ticket <a href="https://github.community/t/if-github-ref-x-while-not-on-the-branch-now-causes-a-build-failure/18055">here</a> for the same issue.</p>
<p>We had reported this issue to the appropriate engineering team for further investigation and evaluation. If they have any progress, I will notify you in time, and sometimes the appropriate engineers may directly reply you here.</p>
<p>You also can follow that ticket and add your comments here.</p>
